src/python/pants/backend/python/lint/ruff/fmt_rules.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Optional
+
+from pants.backend.python.lint.ruff.rules import _run_ruff, _RunRuffRequest
+from pants.backend.python.lint.ruff.subsystem import Ruff, RuffFieldSet, RuffMode
+from pants.backend.python.util_rules import pex
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.core.goals.fmt import AbstractFmtRequest, FmtResult, FmtTargetsRequest
+from pants.core.util_rules.partitions import PartitionerType
+from pants.engine.rules import collect_rules, rule
+from pants.util.logging import LogLevel
+from pants.util.meta import classproperty
+
+
+class RuffFormatRequest(FmtTargetsRequest):
+    field_set_type = RuffFieldSet
+    tool_subsystem = Ruff
+    partitioner_type = PartitionerType.DEFAULT_SINGLE_PARTITION
+
+    @classproperty
+    def tool_name(cls) -> str:
+        return "ruff format"
+
+    @classproperty
+    def tool_id(self) -> str:
+        return "ruff-format"
+
+
+# Note - this function is kept separate because it is invoked from update_build_files.py, but
+# not as a rule.
+async def _run_ruff_fmt(
```

Comment:
maybe this shouldn't be marked internal, as it is being used outside of this module.
```suggestion
async def run_ruff_fmt(
```

Reply:
It is, however I copied the convention that Black and Yapf use for this function - ex: 
* [https://github.com/pantsbuild/pants/blob/d82c578249195494b27e179562c884bc2a7eae14/src/python/pants/backend/python/lint/black/rules.py#L27](Black)
* [https://github.com/pantsbuild/pants/blob/main/src/python/pants/backend/python/lint/yapf/rules.py#L42](Yapf)

I'm not sure if we should change this one to be public (to match standard Python convention) or leave it as is for internal consistency within the repo.

Reply:
let's leave it for consistency then. and if we want to change it, we can do that in a dedicated change cohesively. 👍🏽